tests/test_generics.py

```diff
@@ -1015,3 +1016,33 @@ class Model(GenericModel, Generic[T, U]):
     Model[str, U].__concrete__ is False
     Model[str, U].__parameters__ == [U]
     Model[str, int].__concrete__ is False
+
+
+@skip_36
+def test_generic_recursive_models():
+    T = TypeVar('T')
+
+    class Model1(GenericModel, Generic[T]):
+        ref: 'Model2[T]'
+
+    class Model2(GenericModel, Generic[T]):
+        ref: Union[T, Model1[T]]
+
+    with globals_ctx('Model2', Model2), globals_ctx('T', T):
+        Model1.update_forward_refs()
+        model = Model1[str]
+
+    result = model.parse_obj(dict(ref=dict(ref=dict(ref=dict(ref=123)))))
+    assert result == Model1(ref=Model2(ref=Model1(ref=Model2(ref='123'))))
+
+
+@contextmanager
+def globals_ctx(name: str, value: Any):
```

Comment:
maybe you could use the `create_module` fixture, instead of this magic which is then only used once?

Reply:
fixed 👍